hls-graph/src/Development/IDE/Graph/Internal/Types.hs

```diff
@@ -67,6 +72,12 @@ data SRules = SRules {
 ---------------------------------------------------------------------
 -- ACTIONS
 
+-- | An action representing something that can be run as part of a 'Rule'.
+-- 
+-- 'Action's can be pure functions but also have access to 'IO' via 'MonadIO' and 'MonadUnliftIO.
+-- It should be assumed that actions can throw exceptions, these can be caught with
```

Comment:
Sorry for nitpicking, but who assumes these actions can throw exceptions? The user, or the Rules engine? Or both? E.g., are exceptions always caught or may they bubble up and eventually crash HLS?

Reply:
It's unclear! I'm writing this for the "user of `hls-graph`", which in this case is HLS itself, the point being that `hls-graph` does not do anything about the exceptions, but the user can. Now HLS _does_ use `actionCatch` to catch such exceptions and turn them into diagnostics, but it's a bit of a layering violation to write that here. Perhaps I should say it anyway?

Conceivably some other user of `hls-graph` could choose to just let them bubble up and crash the system...

Reply:
Ah I see, then slightly adapting the beginning of the sentence seems fine by me.

```suggestion
-- It should be assumed that actions throw exceptions, these can be caught with
```

Maybe this is better?